### PR TITLE
[CHAT-1513] partial message update

### PR DESCRIPTION
**Pull request by thesyncim**
# Submit a pull request

## CLA

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [x] The code changes follow best practices
- [x] Code changes are tested (add some information if not applicable)

### CODE REVIEW

**Review by AnatolyRugalev — APPROVED** [review]
LGTM.

We'll sync `pinMessage` shortcuts a bit later

**Review by AnatolyRugalev — DISMISSED** [review]

File: src/types.ts [hunk 1]
```diff
@@ -1709,6 +1710,11 @@ export type PartialUserUpdate<UserType = UnknownType> = {
   unset?: Array<keyof UserResponse<UserType>>;
 };
 
+export type PartialMessageUpdate<MessageType = UnknownType> = {
+  set?: Partial<MessageResponse<MessageType>>;
+  unset?: Array<keyof MessageResponse<MessageType>>;
```
Comment (mahboubii) [inline src/types.ts:1715]: do we allow updates on all the `MessageResponse` properties, like `message.cid`?

Comment (thesyncim) [reply src/types.ts:1715]: no, these are the allowed fields:

- custom data (any field)
- text
- html (server side)
- attachments
- mentioned_user_ids
- quoted_message_id
- silent
- pin

File: src/types.ts [hunk 2]
```diff
@@ -1705,8 +1706,13 @@ export type PartialUpdateChannel<ChannelType = UnknownType> = {
 
 export type PartialUserUpdate<UserType = UnknownType> = {
   id: string;
-  set?: Partial<UserResponse<UserType>>;
-  unset?: Array<keyof UserResponse<UserType>>;
+  set?: Partial<Omit<UserResponse<UserType>, 'cid' | 'created_at' | 'updated_at' | 'deleted_at' |'user' | 'user_id'>>;
+  unset?: Array<keyof Omit<UserResponse<UserType>, 'cid' | 'created_at' | 'updated_at' | 'deleted_at' |'user' | 'user_id'>>;
```
Comment (AnatolyRugalev) [inline src/types.ts:1710]: I guess this should be in `PartialMessageUpdate`, not `PartialUserUpdate`

Comment (AnatolyRugalev) [reply src/types.ts:1710]: 😄

**Review by mahboubii — APPROVED** [review]

File: src/types.ts (hunk 2 above)
Comment (thesyncim) [reply src/types.ts]: 🤦

**Pull request merged**